### PR TITLE
INT-3623: Make `MailSendingMH` more generic

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-3623

**Cherry-pick to 4.1.x**

### CODE REVIEW

File: spring-integration-mail/src/main/java/org/springframework/integration/mail/MailSendingMessageHandler.java [hunk 1]
```diff
@@ -73,12 +74,15 @@ public String getComponentType() {
...
 	protected final void handleMessageInternal(Message<?> message) {
-		MailMessage mailMessage = this.convertMessageToMailMessage(message);
+		MailMessage mailMessage = convertMessageToMailMessage(message);
 		if (mailMessage instanceof SimpleMailMessage) {
 			this.mailSender.send((SimpleMailMessage) mailMessage);
 		}
 		else if (mailMessage instanceof MimeMailMessage) {
-			this.mailSender.send(((MimeMailMessage) mailMessage).getMimeMessage());
+			Assert.state(this.mailSender instanceof JavaMailSender,
+					"'mailSender' [" + this.mailSender + "] doesn't support 'MimeMailMessage' to send: " + mailMessage);
```
Comment (garyrussell) [inline spring-integration-mail/src/main/java/org/springframework/integration/mail/MailSendingMessageHandler.java]: We don't actually know if that is true; there might be some other `MailSender` implementation that __does__ support `MimeMailMessage`, but of course, we can't tell that (and don't know which method to call).

Maybe just change the message to "this adapter requires a JavaMailSender to send a MimeMailMessage" ?

File: spring-integration-mail/src/test/java/log4j.properties [hunk 2]
```diff
@@ -1,8 +1,8 @@
-log4j.rootCategory=DEBUG, stdout
+log4j.rootCategory=WARN, stdout
 
 log4j.appender.stdout=org.apache.log4j.ConsoleAppender
 log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
 log4j.appender.stdout.layout.ConversionPattern=%c{1}: %m%n
 
 log4j.category.org.springframework.integration=WARN
-log4j.category.org.springframework.integration.mail=WARN
+log4j.category.org.springframework.integration.mail=INFO
```
Comment (garyrussell) [inline spring-integration-mail/src/test/java/log4j.properties]: Intentional ?

Comment (artembilan) [inline spring-integration-mail/src/test/java/log4j.properties]: `WARN` - yes, `INFO` - no. :smile:

**Comment by artembilan** [comment]
Pushed. Thank you for pointing that out!

**Comment by artembilan** [comment]
FYI https://github.com/spring-projects/spring-integration-aws/pull/2

**Comment by garyrussell** [comment]
Merged and cherry-picked.

**Pull request closed without merging**